python/docs/source/user_guide/sql/python_udtf.rst

```diff
@@ -0,0 +1,140 @@
+
+.. code-block:: python
+
+    class PythonUDTF:
+
+        def __init__(self) -> None:
+            """
+            Initialize the user-defined table function (UDTF).
+
+            This method is optional to implement and is called once when the UDTF is
+            instantiated. Use it to perform any initialization required for the UDTF.
+            """
+            ...
+
+        def eval(self, *args: Any) -> Iterator[Any]:
+            """"
+            Evaluate the function using the given input arguments.
+
+            This method is required to implement.
+
+            Args:
+                *args: Arbitrary positional arguments representing the input
+                       to the UDTF.
+
+            Yields:
+                tuple: A tuple representing a single row in the UDTF result relation.
+                       Yield thisas many times as needed to produce multiple rows.
+
+            Note:
+                - The result must be a tuple.
+                - UDTFs do not accept keyword arguments on the calling side.
+                - Use "yield" to produce one row at a time for the UDTF result relation,
+                  or use "return" to produce multiple rows for the UDTF result relation at once.
+
+            Example:
+                def eval(self, x: int, y: int):
+                    yield x + y, x - y
+            """
+            ...
+
+        def terminate(self) -> Iterator[Any]:
+            """
+            Called when the UDTF has processed all rows in a partition.
+
+            This method is optional to implement and is useful for performing any
+            cleanup or finalization operations after the UDTF has processed all rows.
+            You can also yield additional rows if needed.
+
+            Yields:
+                tuple: A tuple representing a single row in the UDTF result relation.
+                       Yield this if you want to return additional rows during termination.
+
+            Example:
+                def terminate(self):
+                    yield "done", None
+            """
+            ...
+
+
+The return type of the UDTF must be either a ``StructType`` or a DDL string, both of which 
+define the schema of the UDTF output.
+
+Here's a simple example of a UDTF implementation:
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 39-53
+    :dedent: 4
+
+
+For more detailed usage, please see :func:`udtf`.
+
+
+Registering and Using Python UDTFs in SQL
+-----------------------------------------
+
+Python UDTFs can also be registered and used in SQL queries.
+
+.. literalinclude:: ../../../../../examples/src/main/python/sql/udtf.py
+    :language: python
+    :lines: 58-84
+    :dedent: 4
+
+
+Apache Arrow
+------------
+Apache Arrow is an in-memory columnar data format that is used in Spark to efficiently transfer
+data between JVM and Python processes. Apache Arrow is by default enabled for Python UDTFs.
```

Comment:
It will be disabled by default in 3.5. #42329

Reply:
Yup I will update this doc